src/writer.rs

```diff
@@ -394,4 +458,61 @@ mod indentation {
                 .as_ref()
         );
     }
+    #[test]
+    fn element_writer_empty() {
+        let mut buffer = Vec::new();
+        let mut writer = Writer::new_with_indent(&mut buffer, b' ', 4);
+
+        writer
+            .create_element(b"empty")
+            .with_attribute(("attr1", "value1"))
+            .with_attribute(("attr2", "value2"))
+            .write_empty()
+            .expect("failure");
+
+        assert_eq!(
+            std::str::from_utf8(&buffer).unwrap(),
+            r#"<empty attr1="value1" attr2="value2"/>"#
+        );
+    }
+
+    #[test]
+    fn element_writer_text() {
+        let mut buffer = Vec::new();
+        let mut writer = Writer::new_with_indent(&mut buffer, b' ', 4);
+
+        writer
+            .create_element(b"paired")
+            .with_attribute(("attr1", "value1"))
+            .with_attribute(("attr2", "value2"))
+            .write_text_content(BytesText::from_plain_str("text"))
+            .expect("failure");
+
+        assert_eq!(
+            std::str::from_utf8(&buffer).unwrap(),
+            r#"<paired attr1="value1" attr2="value2">text</paired>"#
+        );
+    }
+
+    #[test]
+    fn element_writer_scope() {
+        let mut buffer = Vec::new();
+        let mut writer = Writer::new(&mut buffer);
+
+        writer
+            .create_element(b"paired")
+            .with_attribute(("attr1", "value1"))
+            .with_attribute(("attr2", "value2"))
+            .write_nested_content(|writer| {
+                writer.create_element(b"inner").write_empty()?;
+                writer.create_element(b"inner2").write_empty()?;
+                Ok(())
+            })
+            .expect("failure");
```

Comment:
I added one new idea.  Maybe this is a step too far?  But it seemed like a neat experiment, and it works.